src/process/process.rs

```diff
@@ -76,15 +75,29 @@ struct LoadInfo {
     bss_end_offset: usize    /* Offset to end of BSS */
 }
 
+struct MemorySlice<T> {
+    data: T,
+    len: usize,
+}
+
+impl<T> MemorySlice<*mut T> {
+    fn from_mut_slice(s: &'static mut [T]) -> MemorySlice<*mut T> {
+        MemorySlice {
+            data: s.as_mut_ptr(),
+            len: s.len(),
+        }
+    }
+}
+
 pub struct Process<'a> {
     /// The process's memory.
-    memory: Slice<u8>,
+    memory: MemorySlice<*mut u8>,
```

Comment:
Main reason I did it this way was to keep the code simpler, and because semantically I couldn't really identify the lifetime of the references. I didn't try using a `'static` lifetime. I'll try that out and see if it ends up looking ok.